src/components/editor/MonacoEditor.tsx

```diff
@@ -22,35 +29,128 @@ function MonacoEditor({ disabled, value, path, height, onChange }: Props) {
         null
     );
     const heightVal = height ?? DEFAULT_HEIGHT;
+    const [isChanging, setIsChanging] = useState<boolean | null>(null);
+    const [hasServerChanges, setHasServerChanges] = useState<any | null>(null);
+    const [showServerDiff, setShowServerDiff] = useState(false);
 
-    const handlers = {
-        change: debounce(async (val) => {
-            if (editorRef.current && onChange) {
-                onChange(JSON.parse(val), path);
+    const serverUpdate = useZustandStore<
+        EditorStoreState<DraftSpecQuery>,
+        EditorStoreState<DraftSpecQuery>['serverUpdate']
+    >((state) => state.serverUpdate);
+
+    useEffect(() => {
+        if (editorRef.current) {
+            const currentStringValue = editorRef.current.getValue();
+
+            if (currentStringValue) {
+                const currentEditorValue = JSON.parse(currentStringValue);
+
+                if (!isEqual(currentEditorValue, serverUpdate)) {
+                    setHasServerChanges(serverUpdate);
+                } else {
+                    setHasServerChanges(null);
+                }
+            }
+        }
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [serverUpdate]);
```

Comment:
Checking if the server updates are something to worry about. This needs to check for ID and other stuff. That'll happen in a follow up.

Reply:
Also... we could look into adding/replacing this with just checking if there were updates before calling the server to save.